kolibri/core/assets/src/api-resource.js

```diff
@@ -237,17 +249,16 @@ class Collection {
         } else {
           this.synced = false;
           this.resource.client({ path: this.url, params }).then((response) => {
-            // Reset current models to only include ones from this fetch.
-            this.models = [];
```

Comment:
You're right, this should be changed. Also, the change below to the `synced` setter makes the `forEach` loop superfluous.